solutionbox/code_free_ml/mltoolbox/code_free_ml/data/trainer/feature_transforms.py

```diff
@@ -0,0 +1,713 @@
+      info[name]['size'] = 1
+    elif transform_name == BOW_TRANSFORM or transform_name == TFIDF_TRANSFORM:
+      info[name + '_ids']['dtype'] = tf.int64
+      info[name + '_weights']['dtype'] = tf.float32
+      info[name + '_ids']['size'] = None
+      info[name + '_weights']['size'] = None
+    elif transform_name == KEY_TRANSFORM:
+      schema_type = [col['type'] for col in schema if col['name'] == name]
+      schema_type = schema_type[0].lower()
+      if schema_type == FLOAT_SCHEMA:
+        info[name]['dtype'] = tf.float32
+      elif schema_type == INTEGER_SCHEMA:
+        info[name]['dtype'] = tf.int64
+      else:
+        info[name]['dtype'] = tf.string
+      info[name]['size'] = 1
+    elif transform_name == TARGET_TRANSFORM:
+      # If the input is a string, it gets converted to an int (id)
+      schema_type = [col['type'] for col in schema if col['name'] == name]
+      schema_type = schema_type[0].lower()
+      if schema_type in NUMERIC_SCHEMA:
+        info[name]['dtype'] = tf.float32
+      else:
+        info[name]['dtype'] = tf.int64
+      info[name]['size'] = 1
+    elif transform_name == IMAGE_TRANSFORM:
+      info[name]['dtype'] = tf.float32
+      info[name]['size'] = 2048
+    else:
+      raise ValueError('Unknown transfrom %s' % transform_name)
+
+  return info
+
+
+# TODO(brandondutra): should numerical defaults be the mean or 0?
+def csv_header_and_defaults(features, schema, keep_target):
+  """Gets csv header and default lists."""
+
+  target_name = None
+  for name, transform in six.iteritems(features):
+    if transform['transform'] == TARGET_TRANSFORM:
+      target_name = name
+      break
+  if keep_target and not target_name:
+    raise ValueError('Cannot find target transform')
+
+  csv_header = []
+  record_defaults = []
+  for col in schema:
+    if not keep_target and col['name'] == target_name:
+      continue
+
+    csv_header.append(col['name'])
+    if col['type'].lower() == 'integer':
+      default, dtype = 0, tf.int64
+    elif col['type'].lower() == 'float':
+      default, dtype = 0.0, tf.float32
+    else:
+      default, dtype = '', tf.string
+
+    record_defaults.append(tf.constant([default], dtype=dtype))
+
+  return csv_header, record_defaults
+
+
+def build_csv_serving_tensors(analysis_path, features, schema, keep_target):
+  """Returns a placeholder tensor and transformed tensors."""
+
+  csv_header, record_defaults = csv_header_and_defaults(features, schema, keep_target)
+
+  placeholder = tf.placeholder(dtype=tf.string, shape=(None,),
+                               name='csv_input_placeholder')
+  tensors = tf.decode_csv(placeholder, record_defaults)
+  raw_features = dict(zip(csv_header, tensors))
+
+  transform_fn = make_preprocessing_fn(analysis_path, features, keep_target)
+  transformed_tensors = transform_fn(raw_features)
+
+  transformed_features = {}
+  for k, v in six.iteritems(transformed_tensors):
+    if isinstance(v, tf.Tensor) and v.get_shape().ndims == 1:
```

Comment:
I copied this from some tft code. But you are right, I don't think so. Tests pass with it removed, so no!